### PR TITLE
Evaluation updated to competing risks

### PR DESCRIPTION
**Pull request by Jeanselme**

### CODE REVIEW

File: dsm/dsm_api.py [hunk 1]
```diff
@@ -120,6 +121,26 @@ def fit(self, x, t, e, vsize=0.15,
     
     return self
 
+  def _eval_conditional_loss(self, x, t, e):
```
Comment (chiragnagpal) [inline dsm/dsm_api.py:124]: _eval_conditional_loss if confusing ..lets call it _eval_nll

**Review by chiragnagpal — COMMENTED** [review]
lets change the name of this function to _eval_nll

File: dsm/dsm_api.py [hunk 2]
```diff
@@ -120,6 +121,26 @@ def fit(self, x, t, e, vsize=0.15,
...
     return self
 
+  def _eval_nll(self, x, t, e):
+    """
+        Evaluates models with conditional loss
+    """
+    if not(self.fitted):
+      raise Exception("The model has not been fitted yet. Please fit the " +
+                      "model using the `fit` method on some training data " +
+                      "before calling `eval`.")
```
Comment (chiragnagpal) [inline dsm/dsm_api.py:131]: change eval to _eval_nll

File: dsm/dsm_api.py [hunk 3]
```diff
@@ -120,6 +121,26 @@ def fit(self, x, t, e, vsize=0.15,
     
     return self
 
+  def _eval_nll(self, x, t, e):
+    """
+        Evaluates models with conditional loss
```
Comment (chiragnagpal) [inline dsm/dsm_api.py:126]: Add more descriptive docstring: specify what x, t , e requires and what the output is. The docstring should clearly describe that this function outputs the negative log likelihood of the dsm model

Comment (chiragnagpal) [reply dsm/dsm_api.py:126]: Also specify how NLL is calculated in the presence of multiple competing risks

File: dsm/dsm_api.py [hunk 4]
```diff
@@ -32,7 +32,8 @@
 from dsm.dsm_torch import DeepConvolutionalSurvivalMachinesTorch
 from dsm.losses import predict_cdf
 import dsm.losses as losses
-from dsm.utilities import train_dsm, _get_padded_features, _get_padded_targets
+from dsm.utilities import train_dsm, _get_padded_features, _get_padded_targets, \
```
Comment (chiragnagpal) [inline dsm/dsm_api.py:35]: break into 3 separate import statements 1) train_dsm, 2) _get_padded_features, _get_padded_targets and 3)    _reshape_tensor_with_nans

**Review by chiragnagpal — CHANGES_REQUESTED** [review]
@Jeanselme  would this work for both simple DSM and Recurrent DSM ? Please check, ideally it should. Also look at the inline comments I have added.

**Comment by Jeanselme** [comment]
Yes, it does

**Comment by chiragnagpal** [comment]
LGTM. Merged.

**Pull request merged**